realm/realm-library/src/main/cpp/io_realm_internal_OsObject.cpp

```diff
@@ -162,18 +162,22 @@ static void finalize_object(jlong ptr)
 
 static inline size_t do_create_row(jlong shared_realm_ptr, jlong table_ptr)
 {
-    auto& shared_realm = *(reinterpret_cast<SharedRealm*>(shared_realm_ptr));
     auto& table = *(reinterpret_cast<realm::Table*>(table_ptr));
-    shared_realm->verify_in_write();
+    if (shared_realm_ptr) {
```

Comment:
You are correct, but `SharedRealm` in `Table` can be `null` and in that case, `0` is passed to this function as a `shared_realm_ptr`.

https://github.com/realm/realm-java/pull/4812/files#diff-cb8c4bdde24368ab202c6405c065a7fdR168

That case only happens when the `Table` is created in our unit tests without `SharedRealm`.
We will be able to remove this check after retirement of old `JNI*Test`.